Mathlib/Data/ENNReal/Operations.lean

```diff
@@ -314,15 +314,37 @@ theorem natCast_sub (m n : ℕ) : ↑(m - n) = (m - n : ℝ≥0∞) := by
 @[deprecated (since := "2024-04-17")]
 alias nat_cast_sub := natCast_sub
 
+/-- See `ENNReal.sub_eq_of_eq_add'` for a version assuming that `a = c + b` itself is finite rather
+than `b`. -/
 protected theorem sub_eq_of_eq_add (hb : b ≠ ∞) : a = c + b → a - b = c :=
   (cancel_of_ne hb).tsub_eq_of_eq_add
 
+/-- Weaker version of `ENNReal.sub_eq_of_eq_add` assuming that `a = c + b` itself is finite rather
+han `b`. -/
```

Comment:
Here and elsewhere, I wouldn't say this is "weaker". The result is *stronger* because the *hypotheses* are weaker.

Reply:
Nono, the result is weaker because the hypothesis is stronger.

If that doesn't convince you, note that I am proving one lemma in terms of the other, the weaker lemma in terms of the stronger